plugins/module_utils/network/nxos/rm_templates/fc_interfaces.py

```diff
@@ -0,0 +1,194 @@
+            "result": {
+                "{{ name }}": {
+                    "description": "{{ description }}",
+                },
+            },
+        },
+        {
+            "name": "enabled",
+            "getval": re.compile(
+                r"""
+                (?P<negate>\s+no)?
+                (?P<shutdown>\s+shutdown)
+                $""", re.VERBOSE,
+            ),
+            "setval": "shutdown",
+            "result": {
+                "{{ name }}": {
+                    "enabled": "{{ False if shutdown is defined and negate is not defined else True }}",
+                },
+            },
+        },
+        {
+            "name": "speed",
+            "getval": re.compile(
+                rf"""\s+switchport\s+speed\s+(?P<speed>{allowed_values_pattern})$""", re.VERBOSE,
+            ),
+            "setval": "switchport speed {{ speed|string }}",
+            "result": {
+                "{{ name }}": {
+                    "speed": "{{ speed|string }}",
+                },
+            },
+        },
+        {
+            "name": "mode",
+            "getval": re.compile(
+                r"""
+                \s+switchport\s+mode\s+(?P<mode>\S+)
+                $""", re.VERBOSE,
+            ),
+            "setval": "switchport mode {{ mode|string }}",
+            "result": {
+                "{{ name }}": {
+                    "mode": "{{ mode }}",
+                },
+            },
+        },
+        {
+            "name": "trunk_mode",
+            "getval": re.compile(
+                r"""
+                \s+switchport\s+trunk\s+mode\s+(?P<trunk_mode>\S+)
+                $""", re.VERBOSE,
+            ),
+            "setval": "switchport trunk mode {{ trunk_mode|string }}",
+            "result": {
+                "{{ name }}": {
+                    "trunk_mode": "{{ trunk_mode }}",
+                },
+            },
+        },
+
+        {
+            "name": "analytics_scsi",
+            "getval": re.compile(
+                r"""
+                (?P<negate>\s+no)?
+                \s+analytics\s+type\s+(?P<analytics_scsi>fc-scsi)
+                $""", re.VERBOSE,
+            ),
+            "setval": "analytics type {{ analytics_scsi|string }}",
+            "result": {
+                "{{ name }}": {
+                    "analytics_scsi": "{{ analytics_scsi }}",
+                },
+            },
+        },
+        {
+            "name": "analytics_nvme",
+            "getval": re.compile(
+                r"""
+                (?P<negate>\s+no)?
+                \s+analytics\s+type\s+(?P<analytics_nvme>fc-nvme)
+                $""", re.VERBOSE,
+            ),
+            "setval": "analytics type {{ analytics_nvme|string }}",
+            "result": {
+                "{{ name }}": {
+                    "analytics_nvme": "{{ analytics_nvme }}",
+                },
+            },
+        },
+
+        {
+            "name": "analytics",
```

Comment:
In which scenario is this parser being used?

Reply:
didnt get you

Reply:
This parser doesn't seem to have a setval. Can you explain that? And also why we need this particular parser when we already have to specific ones above this?

Reply:
we had discussed this over a call regrading this if you remember, because analytics fc-all shows up differently while getting the facts and while configuring its different

Reply:
Oh, looking for closely at the examples, I get that now.